### PR TITLE
Xcdf

### PR DESCRIPTION
**Pull request by ColasRi**
First conda package, need a little help.
It's a small C++ project, so @conda-forge/help-c-cpp would be the right team, with some python 2.7 binding: https://github.com/jimbraun/XCDF
The recipe builds fine locally (MacOS, python2) as far as I can tell.

There are a few warnings like
> WARNING: Disagreement in get_linkages [...]
> WARNING (xcdf,bin/xcdf-simple-test): Needed DSO /usr/lib/libz.1.dylib not found in any CDT/compiler package, nor the whitelist?!
>   INFO (xcdf,bin/xcdf-simple-test): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
> WARNING (xcdf,bin/xcdf-simple-test): Needed DSO /usr/lib/libc++.1.dylib not found in any CDT/compiler package, nor the whitelist?!
>   INFO (xcdf,bin/xcdf-simple-test): lib/libxcdf.dylib found in this package

which I am unsure how to tackle.

But my main question is what is happening with Travis and all the `conda: command not found`: https://travis-ci.com/ColasRi/staged-recipes/builds/102976262#L483

Any help to get on the right tracks would be appreciated!

### CODE REVIEW

File: recipes/xcdf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  url: https://github.com/jimbraun/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 4e445a2fea947ba14505d08177c8d5b55856f8411f28de1fe4d4c00f6824b711
+
+build:
+  number: 0
+  skip: True  # [py3k or win]
+
+requirements:
+  build:
+    # - {{ compiler('c') }}
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]: These lines shouldn't be commented out

File: recipes/xcdf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    # - {{ compiler('cxx') }}
+    - cmake
+    - zlib
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]: This should only be in the `host` or `run` dependencies:
```suggestion
```

**Comment by ColasRi** [comment]
Thanks @chrisburr ! I'll make conda-forge-linter happy now.
BTW, looking forward to the Geant4 recipe.

File: recipes/xcdf/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  number: 0
+  skip: True  # [py3k or win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - zlib
+    - pyhton < 3
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]: The version specification isn't needed as you have the `skip` above. You also probably need `python` as a runtime dependency:
```suggestion
    - python
  run:
    - python
```

File: recipes/xcdf/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{% set name = "XCDF" %}
+{% set version = "v3.00.03" %}
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]:
```suggestion
{% set version = "3.00.03" %}
```

File: recipes/xcdf/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/jimbraun/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]:
```suggestion
  url: https://github.com/jimbraun/{{ name }}/archive/v{{ version }}.tar.gz
```

File: recipes/xcdf/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - zlib
+    - pyhton < 3
+
+test:
+  commands:
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]: From the description I guess there is a importable Python module as well?
```suggestion
  imports:
    - xcdf
  commands:
```

File: recipes/xcdf/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    - bin/xcdf-speed-test
+    - bin/xcdf-seek-test
+    - bin/xcdf-concat-seek-test
+    - bin/xcdf-append-test
+    - bin/xcdf-buffer-fill-test
+    - bin/xcdf-random-test
+
+about:
+  home: https://github.com/jimbraun/XCDF
+  license: custom
```
Comment (chrisburr) [inline recipes/xcdf/meta.yaml]: From a quick glance at the licence it looks like 2-clause BSD, please check:
```suggestion
  license: BSD-2-Clause
  license_family: BSD
```

File: recipes/xcdf/LICENSE [hunk 8]
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2014, University of Maryland
```
Comment (chrisburr) [inline recipes/xcdf/LICENSE]: This file shouldn't be needed as it is included in the tarball.

**Comment by ColasRi** [comment]
Thanks for the review, I accepted all changes.
Good to know for Travis. I was remembering something I read this weekend about the CI sometimes getting stick and needing a flick, I'll be more patient.

**Comment by ColasRi** [comment]
BTW, should I accept proposed changes one by one (with many commits) or re-implement the changes on my side and do a single commit? Not sure what the etiquette is here.

**Comment by chrisburr** [comment]
> BTW, should I accept proposed changes one by one (with many commits) or re-implement the changes on my side and do a single commit? Not sure what the etiquette is here.

Either is fine, the CI is configured to skip outdated commits in pull requests so it makes little difference.

**Comment by chrisburr** [comment]
It looks like it is installing the Python libraries into `$PREFIX/lib/` instead of the site packages directory. The quickest hack would do this after running `make install`:

```bash
mv "${PREFIX}/lib/xcdf" "${SP_DIR}/xcdf"
```

Some potentially helpful links if you want to patch it "properly":

* [You can apply patches before building with conda](https://docs.conda.io/projects/conda-build/en/latest/source/resources/define-metadata.html?highlight=selectors#patches)
* CMake provides this information in the `Python_SITELIB` from [`FindPython`](https://cmake.org/cmake/help/latest/module/FindPython.html#module:FindPython)
* Conda provides this information in `$SP_DIR`

**Comment by ColasRi** [comment]
I'm working on it.
FWIW, things for much easier once I realized that all my local errors were related to working on MacOS (with different than used during the recipe build), and that I could reproduce the CI output when working on a linux environment.
So PR suspended for now until I fix everything.

**Comment by ColasRi** [comment]
I do not have time to look into this in the near future. When I do and have something ready, I will reopen. Thanks for the help so far!

**Pull request closed without merging**